### PR TITLE
AWS OIDC: Move enrollment of EKS clusters to the gRPC service.

### PR DESCRIPTION
**Pull request by AntonAM**
This PR creates a new method on the AWS OIDC gRPC service that enrolls EKS clusters.

This is part of a refactor that moves the API calls behind the Auth Service.

Context: https://github.com/gravitational/teleport/issues/37245

### CODE REVIEW

File: lib/integrations/awsoidc/eks_enroll_clusters.go [hunk 1]
```diff
@@ -235,6 +235,22 @@ type EnrollEKSClustersRequest struct {
 	AgentVersion string
 }
 
+func (e *EnrollEKSClustersRequest) CheckAndSetDefaults() error {
```
Comment (marcoandredinis) [inline lib/integrations/awsoidc/eks_enroll_clusters.go:238]: Missing godoc

File: lib/auth/integration/integrationv1/awsoidc_test.go [hunk 2]
```diff
@@ -180,6 +180,73 @@ func TestListEICE(t *testing.T) {
...
+
+		userCtx := authorizerForDummyUser(t, ctx, role, localClient)
+
+		result, err := awsoidService.EnrollEKSClusters(userCtx, &integrationv1.EnrollEKSClustersRequest{
+			Integration:     integrationName,
+			Region:          "my-region",
+			PublicProxyAddr: proxyPublicAddr,
+			ClusterNames:    []string{"EKS1"},
+		})
+		_ = result
```
Comment (marcoandredinis) [inline lib/auth/integration/integrationv1/awsoidc_test.go:245]:
```suggestion
		_, err := awsoidService.EnrollEKSClusters(userCtx, &integrationv1.EnrollEKSClustersRequest{
			Integration:     integrationName,
			Region:          "my-region",
			PublicProxyAddr: proxyPublicAddr,
			ClusterNames:    []string{"EKS1"},
		})
```

File: lib/auth/integration/integrationv1/awsoidc.go [hunk 3]
```diff
@@ -417,6 +425,57 @@ func (s *AWSOIDCService) DeployDatabaseService(ctx context.Context, req *integra
 	}, nil
 }
 
+func (s *AWSOIDCService) EnrollEKSClusters(ctx context.Context, req *integrationpb.EnrollEKSClustersRequest) (*integrationpb.EnrollEKSClustersResponse, error) {
```
Comment (marcoandredinis) [inline lib/auth/integration/integrationv1/awsoidc.go:428]: Missing godoc

**Review by marcoandredinis — APPROVED** [review]
LGTM, just left a few minors

Can you please use this PR when backporting?
Otherwise, we'll get merge conflits later on
https://github.com/gravitational/teleport/pull/38442

File: api/proto/teleport/integration/v1/awsoidc_service.proto [hunk 4]
```diff
@@ -292,6 +295,47 @@ message DeployServiceResponse {
...
+message EnrollEKSClustersRequest {
+  // Integration is the AWS OIDC Integration name.
+  // Required.
+  string integration = 1;
+  // Region is the AWS Region
+  // Required.
+  string region = 2;
+  // PublicProxyAddr is public address Teleport cluster.
+  // Required.
+  string public_proxy_addr = 3;
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply api/proto/teleport/integration/v1/awsoidc_service.proto]: You're right, change it to use Auth's `getProxyPublicAddr()` https://github.com/gravitational/teleport/pull/38683/commits/d513f41aecdccf6a2724878fe39ad9d1bd3be3db

File: lib/web/integrations_awsoidc.go [hunk 5]
```diff
@@ -434,21 +424,24 @@ func (h *Handler) awsOIDCEnrollEKSClusters(w http.ResponseWriter, r *http.Reques
...
-			IsCloud:            h.ClusterFeatures.GetCloud(),
-			AgentVersion:       agentVersion,
-		})
+	response, err := clt.IntegrationAWSOIDCClient().EnrollEKSClusters(ctx, &integrationv1.EnrollEKSClustersRequest{
+		Integration:        integrationName,
+		Region:             req.Region,
+		ClusterNames:       req.ClusterNames,
+		EnableAppDiscovery: req.EnableAppDiscovery,
+		EnableAutoUpgrades: h.ClusterFeatures.GetAutomaticUpgrades(),
+		IsCloud:            h.ClusterFeatures.GetCloud(),
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/web/integrations_awsoidc.go:433]: Good idea, I think we're fine with `IsCloud`, auto upgrades I can see might be something controllable in the future, but we also can add it in the future if needed 🤔

**Review by tigrato — APPROVED** [review]

File: lib/web/integrations_awsoidc.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/web/integrations_awsoidc.go]: Done https://github.com/gravitational/teleport/pull/38683/commits/972ea692565cd257d703079254ac0a94dc42470f

File: api/proto/teleport/integration/v1/awsoidc_service.proto [hunk 6]
```diff
@@ -292,6 +295,44 @@ message DeployServiceResponse {
...
+  bool is_cloud = 6;
+  // AgentVersion is version of agent Helm chart to install on the EKS cluster.
+  // Required.
+  string agent_version = 7;
+}
+
+// EnrollEKSClusterResult contains result for a single cluster enrollment.
+message EnrollEKSClusterResult {
+  // ClusterName identifies what EKS cluster this result is for.
+  string cluster_name = 1;
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply api/proto/teleport/integration/v1/awsoidc_service.proto]: Done https://github.com/gravitational/teleport/pull/38683/commits/fd805835c3bcc238b6d68ba2f491bb2ab341ac3f

**Pull request merged**